### PR TITLE
Clean up pluginmanager

### PR DESCRIPTION
**Pull request by petemoore**
Some simplifications...

### CODE REVIEW

File: plugins/pluginmanager.go [hunk 1]
```diff
@@ -41,26 +41,24 @@ func mergeErrors(errs ...error) error {
...
 // it'll merge the errors.
-func waitForErrors(errors <-chan error, count int) error {
-	if count == 0 {
-		return nil
+func waitForErrors(taskPlugins []TaskPlugin, f func(p TaskPlugin) error) (err error) {
+	errors := make([]error, len(taskPlugins))
+	var wg sync.WaitGroup
+	for i, j := range taskPlugins {
+		wg.Add(1)
+		go func(i int, j TaskPlugin) {
```
Comment (walac) [inline plugins/pluginmanager.go]: `j` shadows the outer scoped `j`
`i` shadows the outer scoped `i`

You can access the outer variables inside `func`, so you can do
```go
go func () {
```

File: plugins/pluginmanager.go [hunk 2]
```diff
@@ -201,41 +199,39 @@ func (m *pluginManager) PayloadSchema() schematypes.Object {
...
+	taskPlugins := make([]TaskPlugin, len(m.plugins))
+	errors := make([]error, len(m.plugins))
 
-	errors := make(chan error)
+	var wg sync.WaitGroup
 	for i, p := range m.plugins {
-		go func(name string, p Plugin) {
-			taskPlugin, err := p.NewTaskPlugin(TaskPluginOptions{
+		wg.Add(1)
+		go func(i int, p Plugin) {
```
Comment (walac) [inline plugins/pluginmanager.go]: `p` shadows outer scoped `p`
`i` shadows outer scoped `i`

File: plugins/pluginmanager.go [hunk 3]
```diff
@@ -302,17 +286,14 @@ func (m *taskPluginManager) Stopped(r engines.ResultSet) (bool, error) {
...
-			if !success {
-				result.Set(false)
-			}
-			errors <- err
-		}(p)
-	}
-	return result.Get(), waitForErrors(errors, len(m.taskPlugins))
+	err := waitForErrors(m.taskPlugins, func(p TaskPlugin) error {
+		success, err := p.Stopped(r)
+		if !success {
```
Comment (walac) [inline plugins/pluginmanager.go]: is `success` valid if `err != nil`?

File: plugins/pluginmanager.go (hunk 1 above)
Comment (petemoore) [inline plugins/pluginmanager.go]: I'm afraid that's not quite right - you need them, as the go routines run concurrently.

File: plugins/pluginmanager.go (hunk 2 above)
Comment (petemoore) [inline plugins/pluginmanager.go]: See above - this is needed.

File: plugins/pluginmanager.go (hunk 3 above)
Comment (petemoore) [inline plugins/pluginmanager.go]: No idea, this was just a refactor - but this preserves behaviour.

File: plugins/pluginmanager.go (hunk 2 above)
Comment (walac) [inline plugins/pluginmanager.go]: As we discussed in irc, just change parameter names to make gometalinter happy.

File: plugins/pluginmanager.go (hunk 1 above)
Comment (jonasfj) [inline plugins/pluginmanager.go]: yeah, golang doesn't capture variables from `for` loops properly... this is like javascript without `let`

**Comment by petemoore** [comment]
@jonasfj @walac While applying Wander's linting changes, I spotted some more simplifications to make, so the code is even shorter now. Could you review again? Thanks!

File: plugins/pluginmanager.go [hunk 4]
```diff
@@ -247,120 +225,59 @@ func (m *taskPluginManager) Prepare(c *runtime.TaskContext) error {
...
 	defer m.working.Set(false)
 
-	// Run method on plugins in parallel
-	errors := make(chan error)
-	for _, p := range m.taskPlugins {
-		go func(p TaskPlugin) { errors <- p.Prepare(c) }(p)
+	// Execute phase on plugins in parallel
+	errors := make([]error, len(m.taskPlugins))
+	var wg sync.WaitGroup
+	for i, j := range m.taskPlugins {
```
Comment (jonasfj) [inline plugins/pluginmanager.go]: Nit but variables I or index and to or taskPlugin seems more descriptive.

Comment (walac) [inline plugins/pluginmanager.go]: @petemoore is using different variable names to avoid variable shadowing...

**Pull request merged**